feat: add support for password commands in crypt backend

hrntknr opened the pull request:
#### What is the purpose of this change?

crypt backend to allow passwords to be retrieved from the command. This means that passwords can be dynamically retrieved from standard input, secret manager, gpg, etc.

#### Was the change discussed in an issue or in the forum before?

No.

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [x] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [ ] I'm done, this Pull Request is ready for review :-)

albertony commented on docs/content/crypt.md:
Your changes in this file is are in an autogenerated section, so you must make sure the help texts in `backend/crypt/crypt.go` contains everything, and you can omit the changes here in `docs/content/crypt.md` from the PR.

hrntknr replied:
Thanks for your review.
I've integrated all the updates exclusively into backend/crypt/crypt.go. The edits in docs/content/crypt.md have been reverted.

albertony commented:
There is a very similar global flag related to rclone's configuration file:

```
--password-command SpaceSepList       Command for supplying password for encrypted configuration
```

With this implementation:

https://github.com/rclone/rclone/blob/6a7e68aaf2d56df069eb31a46f2f6015a9429b24/fs/config/crypt.go#L173-L205

Did you look at that? Considered re-using the existing function - with some refactoring?

Also, in contrast to that version, your version executes the supplied command in `cmd /c` on Windows and `sh -c` on Unix. I'm not sure that is always the best strategy. E.g. I'm thinking there is a risk of challenges getting the escaping/quoting of arguments correct, or what was your argument for doing that? Wouldn't it be safer and more flexible to not do this, and if required the user can supply a command string which includes the "cmd /c" prefix etc?

hrntknr commented:
Refactored in [8c88d50](https://github.com/rclone/rclone/pull/8450/commits/8c88d5046ac240b463a0543556358d6eda07e745) to use common code with password-command and removed processes like `sh -c` and `cmd /c`.
Please review again.

hrntknr commented:
@albertony Please check again.
